src/Accounts/Authentication.ResourceManager/AzureRmProfile.cs

```diff
@@ -71,10 +71,10 @@ public virtual IAzureContext DefaultContext
                 }
 
                 IAzureContext result = null;
-                if (DefaultContextKey == "Default" && Contexts.Any(c => c.Key != "Default"))
+                if (DefaultContextKey == Constants.DefaultValue && Contexts.Any(c => c.Key != Constants.DefaultValue))
                 {
-                    // If the default context is "Default", but there are other contexts set, remove the "Default" context to throw the below exception
-                    TryCacheRemoveContext("Default");
+                    // If the default context is "Default", but there are other contexts set, remove the "Default" context and select first avaiable context as default
+                    TryRemoveContext(Constants.DefaultValue);
```

Comment:
Debug message might be helpful here
`EnqueueDebugMessage($"Incorrect default context key {DefaultContextKey} found. Trying to remove it and falling back to the first available context.");`